xline/src/storage/db.rs

```diff
@@ -104,22 +95,35 @@ where
         })
     }
 
-    /// Delete key from storage
-    fn delete<K>(&self, table: &'static str, key: K, sync: bool) -> Result<(), ExecuteError>
-    where
-        K: Into<Vec<u8>> + std::fmt::Debug + Sized,
-    {
-        let del_op = WriteOperation::new_delete(table, key.into());
+    fn reset(&self) -> Result<(), ExecuteError> {
+        let start = vec![];
+        let end = vec![0xff];
+        let ops = XLINE_TABLES
+            .iter()
+            .map(|table| WriteOperation::new_delete_range(table, start.as_slice(), end.as_slice()))
+            .collect();
         self.engine
-            .write_batch(vec![del_op], sync)
-            .map_err(|e| ExecuteError::DbError(format!("Failed to delete Lease, error: {e}")))?;
-        Ok(())
+            .write_batch(ops, true)
+            .map_err(|e| ExecuteError::DbError(format!("Failed to reset database, error: {e}")))
     }
 
-    fn write_batch(&self, wr_ops: Vec<WriteOperation>, sync: bool) -> Result<(), ExecuteError> {
-        self.engine
-            .write_batch(wr_ops, sync)
-            .map_err(|e| ExecuteError::DbError(format!("Failed to write batch, error: {e}")))
+    fn put_op(&self, propose_id: &ProposeId, op: WriteOp) {
+        let mut buffer = self.buffer.lock();
+        if let Some(ops) = buffer.get_mut(propose_id) {
+            ops.push(op);
+        } else {
+            let _ignore = buffer.insert(propose_id.clone(), vec![op]);
+        }
+    }
```

Comment:
```suggestion
        buffer.or_insert_with(|| vec![]).push(op);
```

Reply:
it should be `buffer.entry(propose_id.clone()).or_insert_with(|| vec![]).push(op)`, if change to this, `propose_id` will be cloned every time